distribution/kernel-debuginfo/runtest.sh

```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
+set -x
+
+. /usr/bin/rhts_environment.sh
+
+YUM=$(command -v yum)
+if [ -z "$YUM" ]
+then
+	YUM=$(command -v dnf)
+fi
+
+function test_pass()
+{
+	report_result $TEST PASS 0
+}
+
+function test_fail()
+{
+	report_result $TEST FAIL 1
+}
+
+function test_skip()
+{
+	report_result $TEST SKIP 0
+}
+
+function nvr_kernel_current()
+{
+	rpm -qa "kernel*" | grep "$(uname -r)" | sort | head -n1
+}
+
+function nvr_kernel_latest()
+{
+	$YUM list --available kernel{,-rt}.$(uname -m) --showduplicates \
+	| sed "s/.$(uname -m)//" \
+	| sort -k1,1 -k2V \
+	| awk 'END { print $1"-"$2".'$(uname -m)'"; }'
+}
+
+# Determine which kernel debuginfo to download.
+# By default, use currently running kernel (unless overriden from env).
+# This can be achieved either by specifying KERNEL_NVR directly, or one
+# of the following two test config options:
+# CONFIG_NVR_KERNEL_{CURRENT,LATEST}
+
+if [ -z "${KERNEL_NVR++}" ]
+then
+	echo "DEBUG: Using current kernel nvr."
+	KERNEL_NVR="$(nvr_kernel_current)"
+else
+	echo "DEBUG: Using user provided kernel NVR: $KERNEL_NVR"
+fi
+
+if [ -n "$CONFIG_NVR_KERNEL_CURRENT" ]
+then
+	echo "DEBUG: Using current kernel nvr."
+	KERNEL_NVR="$(nvr_kernel_current)"
+fi
+
+if [ -n "$CONFIG_NVR_KERNEL_LATEST" ]
+then
+	echo "DEBUG: Using latest available kernel."
+	KERNEL_NVR="$(nvr_kernel_latest)"
+fi
+
+if [ -z "${KERNEL_NVR++}" ]
+then
+	echo "ERROR: Unable to determine kernel nvr."
+	echo "DEBUG: CONFIG_NVR_KERNEL_CURRENT: " \
+	     $CONFIG_NVR_KERNEL_CURRENT
+	echo "DEBUG: CONFIG_NVR_KERNEL_LATEST: " \
+	     $CONFIG_NVR_KERNEL_LATEST
+	echo "DEBUG: rpm -qa \"kernel*\""
+	rpm -qa "kernel*"
+	echo "DEBUG: uname -r: $(uname -r)"
+	echo "DEBUG: $YUM list --available kernel{,rt}.$(uname -m)" \
+	     "--showduplicates:"
+	$YUM list --available kernel{,-rt}.$(uname -m) --showduplicates
+	test_skip "FAILED. SEE ERROR OUTPUT ABOVE."
+	exit 1
```

Comment:
Restraint interprets non-zero exit codes as a failure and thus this overrides the SKIP. For skip to propagate correctly the exit codes needs to be 0.

Reply:
Changed in v3.